pkg/asset/installconfig/aws/validation.go

```diff
@@ -80,8 +95,18 @@ func validatePlatform(ctx context.Context, meta *Metadata, fldPath *field.Path,
 
 func validateAMI(ctx context.Context, config *types.InstallConfig) field.ErrorList {
 	// accept AMI from the rhcos stream metadata
-	if sets.NewString(rhcos.AMIRegions...).Has(config.Platform.AWS.Region) {
-		return nil
+	switch config.ControlPlane.Architecture {
+	case types.ArchitectureAMD64:
+		if sets.NewString(rhcos.AMIRegionsX86_64...).Has(config.Platform.AWS.Region) {
+			return nil
+		}
+	case types.ArchitectureARM64:
+		if sets.NewString(rhcos.AMIRegionsAARCH64...).Has(config.Platform.AWS.Region) {
+			return nil
+		}
+	default:
+		errMsg := fmt.Sprintf("Architecture: %s is not supported", config.ControlPlane.Architecture)
+		return field.ErrorList{field.Required(field.NewPath("platform", "aws", "amiID"), errMsg)}
```

Comment:
The error message talks about architecture, but the field that the error points to is the AMI ID.
```suggestion
		return field.ErrorList{field.NotSupported(field.NewPath("controlPlane", "architecture"), config.ControlPlane.Architecture, ...)}
```